e2e/tests/upgrades/upgrade_test.go

```diff
@@ -747,7 +751,34 @@ func (s *UpgradeTestSuite) TestV7ToV8ChainUpgrade() {
 
 	t.Run("upgrade chain", func(t *testing.T) {
 		govProposalWallet := s.CreateUserOnChainB(ctx, testvalues.StartingTokenAmount)
-		s.UpgradeChain(ctx, chainB, govProposalWallet, testCfg.UpgradeConfig.PlanName, testCfg.ChainConfigs[0].Tag, testCfg.UpgradeConfig.Tag)
+		s.upgradeChainWithInFlightProposals(ctx, chainB, govProposalWallet, testCfg.UpgradeConfig.PlanName, testCfg.ChainConfigs[0].Tag, testCfg.UpgradeConfig.Tag)
+	})
+
+	t.Run("legacy IBC software upgrade proposal fails", func(t *testing.T) {
```

Comment:
the race conditions here are not great :confused: 

We need to submit a proposal after the upgrade proposal, but before the upgrade and before the voting period elapses. I attempted to side step this by putting the proposal into the deposit period and then activating it after the upgrade